File: TDD Boot Camp/Part 01/Code/OOPrinciples/src/com/oop/encapsulation/Car.java
package com.oop.encapsulation;

public class Car {
	private Engine engine;
	Car(Engine engine){
		this.engine = engine;
	}
	
    public void startEngine() {
       engine.start();
    }


    public void shutDownEngine() {
    	engine.start();
    }
}

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Encapsulation/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Encapsulation
{
    public class Car
    {
        private Engine _engine;
        public Car(Engine engine)
        {
            _engine = engine;
        }

        public void StartEngine()
        {
            _engine.Start();
        }

        public void ShutDownEngine()
        {
            _engine.Start();
        }
    }
}
